test/core/util/resolve_localhost_ip46.cc

```diff
@@ -59,4 +59,20 @@ void LocalhostResolves(bool* ipv4, bool* ipv6) {
   *ipv6 = localhost_to_ipv6;
 }
 
+bool RunningWithIPv6Only() {
+  bool localhost_resolves_to_ipv4 = false;
+  bool localhost_resolves_to_ipv6 = false;
+  LocalhostResolves(&localhost_resolves_to_ipv4, &localhost_resolves_to_ipv6);
+  return !localhost_resolves_to_ipv4 && localhost_resolves_to_ipv6;
+}
+
+absl::string_view LocalIp() {
+  return RunningWithIPv6Only() ? "[::1]" : "127.0.0.1";
+}
+
+std::string LocalIpUri(int port) {
```

Comment:
Most calls to this function look like this:
```
grpc_core::URI::Parse(grpc_core::LocalIpUri(port))
```
I would suggest having a function for that to reduce the code duplication.

Reply:
Only 5 of 38 calls use that pattern.  I don't think that's worth further refactoring.